public/app/plugins/datasource/prometheus/result_transformer.ts

```diff
@@ -293,15 +293,17 @@ function getDataLinks(options: ExemplarTraceIdDestination): DataLink[] {
     const dataSourceSrv = getDataSourceSrv();
     const dsSettings = dataSourceSrv.getInstanceSettings(options.datasourceUid);
 
-    dataLinks.push({
-      title: options.urlDisplayLabel || `Query with ${dsSettings?.name}`,
-      url: '',
-      internal: {
-        query: { query: '${__value.raw}', queryType: 'traceId' },
-        datasourceUid: options.datasourceUid,
-        datasourceName: dsSettings?.name ?? 'Data source not found',
-      },
-    });
+    if (dsSettings) {
```

Comment:
Lets just add a comment what are the cases when this is undefined ie:
- permissions
- deleted datasource
- misconfiguration
And that permission is probably most likely and in that case it makes sense to just not show the link